File: src/main/java/com/Backend/music_app/dto/request/SongCreateRequest.java
package com.Backend.music_app.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class SongCreateRequest {

    String title;

    // Chỉ nhận ID (chuỗi hoặc số) từ Frontend gửi xuống
    UUID artistId;
    @JsonProperty("album_id")
    UUID albumId;
    @JsonProperty("genre_id")
    UUID genreId;

    int duration;
    String audioUrlNormal;
    String audioUrlHq;
    String coverUrl;
    String lyricsPlain;

}

File: lib/data/createRequest/SongCreate_request.dart
class SongCreateRequest {
  String title;
  String artistId;
  String albumId;
  String genreId;
  int duration;
  String audioUrlNormal;
  String audioUrlHq;
  String coverUrl;
  String lyricsPlain;



  SongCreateRequest({
    required this.title,
    required this.artistId,
    required this.albumId,
    required this.genreId,
    required this.duration,
    required this.audioUrlNormal,
    required this.audioUrlHq,
    required this.coverUrl,
    required this.lyricsPlain,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'artistId': artistId,
      'albumId': albumId,
      'genreId': genreId,
      'duration': duration,
      'audioUrlNormal': audioUrlNormal,
      'audioUrlHq': audioUrlHq,
      'coverUrl': coverUrl,
      'lyricsPlain': lyricsPlain,
    };
  }


}